pyaedt/modeler/Primitives3DLayout.py

```diff
@@ -54,9 +54,14 @@ def __getitem__(self, partname):
           Part object details.
 
         """
-        for el in self.geometries:
-            if el == partname:
-                return self.geometries[el]
+        if partname in self.geometries:
+            return self.geometries[partname]
+        if partname in self.vias:
+            return self.nets[partname]
```

Comment:
@maxcapodi78 should it return self.vias

```suggestion
            return self.vias[partname]
```